kolibri/core/notifications/models.py

```diff
@@ -0,0 +1,115 @@
+
+    If using sqlite, this command must be executed to run migrations under this router:
+        `kolibri manage migrate notifications --database=notifications_db`
+    """
+
+    def db_for_read(self, model, **hints):
+        """Send all read operations on Notifications app models to `notifications_db`."""
+        if model._meta.app_label == 'notifications':
+            return 'notifications_db'
+        return None
+
+    def db_for_write(self, model, **hints):
+        """Send all write operations on Notifications app models to `notifications_db`."""
+        if model._meta.app_label == 'notifications':
+            return 'notifications_db'
+        return None
+
+    def allow_relation(self, obj1, obj2, **hints):
+        """Determine if relationship is allowed between two objects."""
+
+        # Allow any relation between two models that are both in the Notifications app.
+        if obj1._meta.app_label == 'notifications' and obj2._meta.app_label == 'notifications':
+            return True
+        # No opinion if neither object is in the Notifications app (defer to default or other routers).
+        elif 'notifications' not in [obj1._meta.app_label, obj2._meta.app_label]:
+            return None
+
+        # Block relationship if one object is in the Notifications app and the other isn't.
+        return False
+
+    def allow_migrate(self, db, app_label, model_name=None, **hints):
+        """Ensure that the Notifications app's models get created on the right database."""
+        if app_label == 'notifications':
+            # The Notifications app should be migrated only on the notifications_db database.
+            return db == 'notifications_db'
+        elif db == 'notifications_db':
+            # Ensure that all other apps don't get migrated on the notifications_db database.
+            return False
+
+        # No opinion for all other scenarios
+        return None
+
+
+class myEnum(object):
+    values = dict()
+
+    @classmethod
+    def choices(cls):
+        return tuple([(name, value) for name, value in sorted(cls.values.items())])
+
+
+class NotificationType(myEnum):
+    Resource = 'Resource'
+    Quiz = 'Quiz'
+    Help = 'Help'
+    Lesson = 'Lesson'
+    values = {'Resource': 'ResourceIndividualCompletion',
+              'Quiz': 'QuizIndividualCompletion',
+              'Help': 'LessonResourceIndividualNeedsHelpEvent',
+              'Lesson': 'LessonIndividualCompletion'}
+
+
+class HelpReason(myEnum):
+    Multiple = 'Multiple'
+    values = {'Multiple': 'MultipleUnsuccessfulAttempts'}
+
+
+@python_2_unicode_compatible
+class LearnerProgressNotification(models.Model):
+    id = UUIDField(primary_key=True)
+    notification_type = models.CharField(max_length=200, choices=NotificationType.choices(), blank=True)
+    user_id = UUIDField()
+    classroom_id = UUIDField()
+    contentnode_id = UUIDField(null=True)
+    contentnode_id = UUIDField(null=True)
```

Comment:
Is it a typo that this line is duplicated?

It might be helpful to store the content ID in addition to the node ID.

Reply:
It was a typo. It's fixed now. According to the specs, the frontend only needs the node id, it does not use the content id (if needed, adding channel id could make sense as well)